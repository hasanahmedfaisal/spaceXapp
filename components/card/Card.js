import React from 'react'
import isEmpty from 'lodash/isEmpty'

export default class Card extends React.Component {
  fetchImageSrc(missionPatchImage) {
    try {
      if (missionPatchImage == null) {
        return ''
      } else if (this.canUseWebP()) {
        return `/${missionPatchImage.split('/')[5].split('.')[0]}.webp`
      } else {
        return missionPatchImage
      }
    } catch (error) {
      return missionPatchImage
    }
  }

  render() {
    const missionName = this.props.cardProps.mission_name
    const flightNumber = this.props.cardProps.flight_number
    const launchYear = this.props.cardProps.launch_year
    const launchSuccess = this.props.cardProps.launch_success
    const landSuccess = this.props.cardProps.land_success
    const missionPatchImage = this.props.cardProps.mission_patch_small
    const missionId = this.props.cardProps.mission_id
    const isMissionIdPresent = !isEmpty(missionId)
    const imageSrc = this.fetchImageSrc(missionPatchImage)
    return (
      <div className='mainCard'>
        <picture>
          <source srcSet={imageSrc} type='image/webp' />
          <img alt='Image Unavailable' src={missionPatchImage} />
        </picture>
        <div className='text missionName'>{missionName} #{flightNumber}</div>
        <div className='text missionId'>Mission Ids:
          <div className='missionIds'>{isMissionIdPresent ? missionId.join() : 'Unavailable'}</div>
        </div>
        <div className='text launchYr'>Launch Year: {launchYear}</div>
        <div className='text launchStatus'>Successful Launch: {launchSuccess ? 'True' : 'False'}</div>
        <div className='text landingStatus'>Successful Landing: {landSuccess ? 'True' : 'False'}</div>
      </div>
    )
  }
}
