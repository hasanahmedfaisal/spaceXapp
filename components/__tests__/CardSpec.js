import Card from '../card/Card'
import renderer from 'react-test-renderer'

describe('Card Filter', () => {
  it('should render correctly', () => {
    const cardProps = {
      cardProps: {
        mission_name: 'some mission name',
        flight_number: 'some flight name',
        launch_year: 'some launch year',
        launch_success: true,
        land_success: true,
        mission_patch_small: 'some url',
        mission_id: ['123']
      }
    }
    const tree = renderer.create(<Card {...cardProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
