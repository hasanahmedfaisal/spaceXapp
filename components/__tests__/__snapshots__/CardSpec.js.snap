// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Filter should render correctly 1`] = `
<div
  className="mainCard"
>
  <picture>
    <source
      srcSet="some url"
      type="image/webp"
    />
    <img
      alt="Image Unavailable"
      src="some url"
    />
  </picture>
  <div
    className="text missionName"
  >
    some mission name
     #
    some flight name
  </div>
  <div
    className="text missionId"
  >
    Mission Ids:
    <div
      className="missionIds"
    >
      123
    </div>
  </div>
  <div
    className="text launchYr"
  >
    Launch Year: 
    some launch year
  </div>
  <div
    className="text launchStatus"
  >
    Successful Launch: 
    True
  </div>
  <div
    className="text landingStatus"
  >
    Successful Landing: 
    True
  </div>
</div>
`;
